### PR TITLE
Overhaul to Python Package/Environment Management

### PR DESCRIPTION
**Pull request by TheSkyentist**
I've added pixi, but as well I tried to add a bit more intro and explanation to all of the tools listed here, as well as pros and cons to each section. Happy to iterate on this.

### CODE REVIEW

File: faq/chapters/python/virtualenv.md
[the remark this answers is not in the document]
Comment (mfouesneau) [reply faq/chapters/python/virtualenv.md:1]: I would move the mamba to right after venv for ease with known techs. Poetry then as established tool then the others. as uv and pixi are new and evolving it's maybe best to have them last.

**Review by mfouesneau — CHANGES_REQUESTED** [review]

File: faq/chapters/python/virtualenv.md
[the remark this answers is not in the document]
Comment (TheSkyentist) [reply faq/chapters/python/virtualenv.md:1]: I've moved the mamba section :)

File: faq/chapters/python/virtualenv.md [hunk 1]
```diff
@@ -1,113 +1,154 @@
...
-To deactivate a virtual environment, type:
+To deactivate a virtual environment, simply type:
 ```bash
 deactivate
 ```
 
-```{attention}
-The `venv` module does not offer to install a different python version than the one used to create the environment. If you need to use a different version of Python, consider using the other options below.
-```
-
```
[the remark this answers is not in the document]
Comment (TheSkyentist) [reply faq/chapters/python/virtualenv.md:62]: I've added it back in

File: faq/chapters/python/virtualenv.md
[the remark this answers is not in the document]
Comment (TheSkyentist) [reply faq/chapters/python/virtualenv.md:1]: Not sure why the diff is so bad, might be worth checking the branch out and reading through the direct version?

File: faq/chapters/python/virtualenv.md [hunk 2]
```diff
@@ -122,56 +163,113 @@ To add a package to your project, use the following command:
...
 ```
 
-more information on poetry can be found in the [official documentation](https://python-poetry.org/docs/).
+More information on poetry can be found in the [official documentation](https://python-poetry.org/docs/). It comes with a lot of great features, include the ability to directly publish your package to PyPI! 
+
+Pros: In terms of project-scoped environment management, it's hard to think of what features would be better suited for reproducibility than what is provided by Poetry. If you are developing a Python package, Poetry is a great option. 
+
+Cons: Some parts of Poetry can be slightly slow and it also cannot specify non-Python dependencies (other than Python itself). 
+
+## `mamba` (replacement for `conda`)
```
[the remark this answers is not in the document]
Comment (TheSkyentist) [reply faq/chapters/python/virtualenv.md]: I think this is a diff issue, I merged the two sections and made it more explicit between micromamba and mamba and that the former is preferred.

**Review by mfouesneau — APPROVED** [review]

**Pull request merged**